File: rxfetch/src/test/java/github/hotstu/labo/rxfetch/T.java
package github.hotstu.labo.rxfetch;

import org.junit.Test;

import io.reactivex.Observable;

/**
 * @author hglf [hglf](https://github.com/hotstu)
 * @desc
 * @since 7/31/19
 */
public class T {
    @Test
    public void testRx() {
        Observable<Integer> ob = Observable.create(emitter -> {
            System.out.println("-----1-----");
            emitter.onNext(1);
            emitter.onComplete();
        });

        Observable<String> objectObservable = ob.flatMap(i -> {
            System.out.println("-----2-----");
            return Observable.just("-" + i + "-");
        });

        objectObservable.subscribe(s -> {
            System.out.println(s);
        });

    }
}

File: noo/src/test/java/github/hotstu/labo/noo/T.kt
package github.hotstu.labo.noo

import android.graphics.Rect
import io.reactivex.Observable
import org.junit.Test

/**
 * @author hglf [hglf](https://github.com/hotstu)
 * @since 7/9/19
 * @desc
 */
class T {
    fun ltwh(left: Int, top: Int, w: Int, h: Int): Rect {
        return Rect(left, top, left + w, top + h)
    }
    @Test
    fun main() {
        Observable.create<String> {
            //it.onNext("1")
            it.onComplete()
        }.flatMap { t: String -> Observable.just("") }.subscribe({ println(it) }, {  }, { println("complete")})


    }
}